src/analysis/analytic/composition.lean

```diff
@@ -104,6 +104,28 @@ begin
   rw [fin.ext_iff, fin.coe_cast_le, composition.ones_embedding, fin.coe_mk],
 end
 
+lemma apply_composition_single (p : formal_multilinear_series 𝕜 E F) {n : ℕ} (hn : 0 < n)
+  (v : fin n → E) : p.apply_composition (composition.single n hn) v = λ j, p n v :=
+begin
+  ext j,
+  refine p.congr (by simp) (λ i hi1 hi2, _),
+  dsimp,
+  congr' 1,
+  convert composition.single_embedding hn ⟨i, hi2⟩,
+  cases j,
+  have : j_val = 0 := le_bot_iff.1 (nat.lt_succ_iff.1 j_property),
+  unfold_coes,
+  congr; try { assumption <|> simp },
```

Comment:
Would it be useful to extract a lemma like,
```
lemma foo 
  {n : ℕ} (hn : 0 < n)
  (j₁ j₂ : fin (composition.single n hn).length)   (hj: j₁.val = j₂.val)
  (i : ℕ) (hi₁ : i < (composition.single n hn).blocks_fun j₁)
  (hi₂ : i < (composition.single n hn).blocks_fun j₂) :
  ((composition.single n hn).embedding j₁) ⟨i, hi₁⟩ =
    ((composition.single n hn).embedding j₂) ⟨i, hi₂⟩
```

Reply:
I am not convinced it would be used more than once, so I left it as it is for now. If I realize that in fact this is useful several times, I will extract the lemma.

Reply:
Sounds good!